src/App.tsx

```diff
@@ -1,8 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+import * as api from './api';
+import PostsList from './PostList';
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadData = async () => {
+    setIsLoading(true);
+
+    const postsFromServer = await api.getPosts();
+
+    setPosts(postsFromServer);
+    setIsLoading(false);
+    setIsLoaded(true);
+  };
+
+  return (
+    <>
+      <h1>Dynamic list of posts</h1>
+      {isLoaded ? (
+        <PostsList posts={posts} />
+      ) : (
```

Comment:
Here at the HTML we can and the error const to give the user the information in case there is an error.
So between lines 27 and 28 we can add `{error && <p>{error}</p>}`